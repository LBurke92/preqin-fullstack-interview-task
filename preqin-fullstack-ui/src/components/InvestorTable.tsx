import {Table} from "@radix-ui/themes"
import {Investor} from "@/types"
import "@radix-ui/themes/styles.css"

interface Props {
  investors: Investor[]
  selectInvestorName?: (investorName: string) => void
  selectCommitmentAssetClass?: (commitmentAssetClass: string) => void
}

export default function InvestorTable(props: Props) {
  const handleSelectedInvestorName = async (investor: Investor) => {
    if (props.selectInvestorName) props.selectInvestorName(investor.investorName)
  }

  const handleSelectedCommitmentAssetClass = async (investor: Investor) => {
    if (props.selectCommitmentAssetClass) props.selectCommitmentAssetClass(investor.commitmentAssetClass)
  }

  return (
    <Table.Root layout="auto" size="1">
      <Table.Header>
        <Table.Row>
          <Table.ColumnHeaderCell align="left">ID</Table.ColumnHeaderCell>
          <Table.ColumnHeaderCell align="left">Investor Name</Table.ColumnHeaderCell>
          <Table.ColumnHeaderCell align="left">Investory Type</Table.ColumnHeaderCell>
          <Table.ColumnHeaderCell align="left">Investor Country</Table.ColumnHeaderCell>
          <Table.ColumnHeaderCell align="left">Investor Date Added</Table.ColumnHeaderCell>
          <Table.ColumnHeaderCell align="left">Investor Last Updated</Table.ColumnHeaderCell>
          <Table.ColumnHeaderCell align="left">Commitment Asset Class</Table.ColumnHeaderCell>
          <Table.ColumnHeaderCell align="left">Commitmment Amount</Table.ColumnHeaderCell>
          <Table.ColumnHeaderCell align="left">Commitmment Currency</Table.ColumnHeaderCell>
        </Table.Row>
      </Table.Header>
      <Table.Body>
        {props.investors.map(investor => {
          return (
            <Table.Row key={investor.id}>
              <Table.RowHeaderCell>{investor.id}</Table.RowHeaderCell>
              <Table.Cell data-testid={investor.investorName} onClick={() => handleSelectedInvestorName(investor)}>
                {investor.investorName}
              </Table.Cell>
              <Table.Cell>{investor.investoryType}</Table.Cell>
              <Table.Cell>{investor.investorCountry}</Table.Cell>
              <Table.Cell>{investor.investorDateAdded}</Table.Cell>
              <Table.Cell>{investor.investorLastUpdated}</Table.Cell>
              <Table.Cell
                data-testid={investor.commitmentAssetClass}
                onClick={() => handleSelectedCommitmentAssetClass(investor)}
              >
                {investor.commitmentAssetClass}
              </Table.Cell>
              <Table.Cell>{investor.commitmentAmount}</Table.Cell>
              <Table.Cell>{investor.commitmentCurrency}</Table.Cell>
            </Table.Row>
          )
        })}
      </Table.Body>
    </Table.Root>
  )
}
