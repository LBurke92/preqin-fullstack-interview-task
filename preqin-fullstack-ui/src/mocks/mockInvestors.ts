export const mockAllInvestors = [
  {
      "investorName": "Ioo Gryffindor fund",
      "investoryType": "fund manager",
      "investorCountry": "Singapore",
      "investorDateAdded": "2000-07-06",
      "investorLastUpdated": "2024-02-21",
      "commitmentAssetClass": "Infrastructure",
      "commitmentAmount": 15000000,
      "commitmentCurrency": "GBP",
      "id": 0
  },
  {
      "investorName": "Ibx Skywalker ltd",
      "investoryType": "asset manager",
      "investorCountry": "United States",
      "investorDateAdded": "1997-07-21",
      "investorLastUpdated": "2024-02-21",
      "commitmentAssetClass": "Infrastructure",
      "commitmentAmount": 31000000,
      "commitmentCurrency": "GBP",
      "id": 1
  },
  {
      "investorName": "Cza Weasley fund",
      "investoryType": "wealth manager",
      "investorCountry": "United Kingdom",
      "investorDateAdded": "2002-05-29",
      "investorLastUpdated": "2024-02-21",
      "commitmentAssetClass": "Hedge Funds",
      "commitmentAmount": 58000000,
      "commitmentCurrency": "GBP",
      "id": 2
  },
  {
      "investorName": "Mjd Jedi fund",
      "investoryType": "bank",
      "investorCountry": "China",
      "investorDateAdded": "2010-06-08",
      "investorLastUpdated": "2024-02-21",
      "commitmentAssetClass": "Private Equity",
      "commitmentAmount": 72000000,
      "commitmentCurrency": "GBP",
      "id": 3
  },
  {
      "investorName": "Mjd Jedi fund",
      "investoryType": "bank",
      "investorCountry": "China",
      "investorDateAdded": "2010-06-08",
      "investorLastUpdated": "2024-02-21",
      "commitmentAssetClass": "Natural Resources",
      "commitmentAmount": 1000000,
      "commitmentCurrency": "GBP",
      "id": 4
  },
  {
      "investorName": "Ibx Skywalker ltd",
      "investoryType": "asset manager",
      "investorCountry": "United States",
      "investorDateAdded": "1997-07-21",
      "investorLastUpdated": "2024-02-21",
      "commitmentAssetClass": "Real Estate",
      "commitmentAmount": 17000000,
      "commitmentCurrency": "GBP",
      "id": 5
  },
  {
      "investorName": "Ibx Skywalker ltd",
      "investoryType": "asset manager",
      "investorCountry": "United States",
      "investorDateAdded": "1997-07-21",
      "investorLastUpdated": "2024-02-21",
      "commitmentAssetClass": "Real Estate",
      "commitmentAmount": 83000000,
      "commitmentCurrency": "GBP",
      "id": 6
  },
  {
      "investorName": "Mjd Jedi fund",
      "investoryType": "bank",
      "investorCountry": "China",
      "investorDateAdded": "2010-06-08",
      "investorLastUpdated": "2024-02-21",
      "commitmentAssetClass": "Hedge Funds",
      "commitmentAmount": 28000000,
      "commitmentCurrency": "GBP",
      "id": 7
  },
  {
      "investorName": "Ibx Skywalker ltd",
      "investoryType": "asset manager",
      "investorCountry": "United States",
      "investorDateAdded": "1997-07-21",
      "investorLastUpdated": "2024-02-21",
      "commitmentAssetClass": "Hedge Funds",
      "commitmentAmount": 85000000,
      "commitmentCurrency": "GBP",
      "id": 8
  },
  {
      "investorName": "Ioo Gryffindor fund",
      "investoryType": "fund manager",
      "investorCountry": "Singapore",
      "investorDateAdded": "2000-07-06",
      "investorLastUpdated": "2024-02-21",
      "commitmentAssetClass": "Hedge Funds",
      "commitmentAmount": 31000000,
      "commitmentCurrency": "GBP",
      "id": 9
  }
]