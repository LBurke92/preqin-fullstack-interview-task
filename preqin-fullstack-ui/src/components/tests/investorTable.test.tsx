import {render, screen} from "@testing-library/react"
import InvestorTable from "@/components/InvestorTable"
import {mockAllInvestors} from "@/mocks/mockInvestors"

describe("InvestorTable", () => {
  it("matches the DOM snapshot", () => {
    const component = render(<InvestorTable investors={mockAllInvestors} />)

    expect(component).toMatchSnapshot()
  })
  it("displays x", () => {
    render(<InvestorTable investors={mockAllInvestors} />)

    expect(screen.getAllByTestId("Ioo Gryffindor fund")[0]).toBeInTheDocument()
  })
})
