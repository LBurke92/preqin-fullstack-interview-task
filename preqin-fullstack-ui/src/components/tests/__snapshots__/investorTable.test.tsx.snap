// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InvestorTable > matches the DOM snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rt-TableRoot rt-r-size-1 rt-variant-ghost"
      >
        <div
          class="rt-ScrollAreaRoot"
          dir="ltr"
          style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
        >
          <style>
            [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
          </style>
          <div
            class="rt-ScrollAreaViewport"
            data-radix-scroll-area-viewport=""
            style="overflow-x: scroll; overflow-y: scroll;"
          >
            <div
              style="min-width: 100%; display: table;"
            >
              <table
                class="rt-TableRootTable rt-r-tl-auto"
              >
                <thead
                  class="rt-TableHeader"
                >
                  <tr
                    class="rt-TableRow"
                  >
                    <th
                      align="left"
                      class="rt-TableCell rt-TableColumnHeaderCell"
                      scope="col"
                    >
                      ID
                    </th>
                    <th
                      align="left"
                      class="rt-TableCell rt-TableColumnHeaderCell"
                      scope="col"
                    >
                      Investor Name
                    </th>
                    <th
                      align="left"
                      class="rt-TableCell rt-TableColumnHeaderCell"
                      scope="col"
                    >
                      Investory Type
                    </th>
                    <th
                      align="left"
                      class="rt-TableCell rt-TableColumnHeaderCell"
                      scope="col"
                    >
                      Investor Country
                    </th>
                    <th
                      align="left"
                      class="rt-TableCell rt-TableColumnHeaderCell"
                      scope="col"
                    >
                      Investor Date Added
                    </th>
                    <th
                      align="left"
                      class="rt-TableCell rt-TableColumnHeaderCell"
                      scope="col"
                    >
                      Investor Last Updated
                    </th>
                    <th
                      align="left"
                      class="rt-TableCell rt-TableColumnHeaderCell"
                      scope="col"
                    >
                      Commitment Asset Class
                    </th>
                    <th
                      align="left"
                      class="rt-TableCell rt-TableColumnHeaderCell"
                      scope="col"
                    >
                      Commitmment Amount
                    </th>
                    <th
                      align="left"
                      class="rt-TableCell rt-TableColumnHeaderCell"
                      scope="col"
                    >
                      Commitmment Currency
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="rt-TableBody"
                >
                  <tr
                    class="rt-TableRow"
                  >
                    <th
                      class="rt-TableCell rt-TableRowHeaderCell"
                      scope="row"
                    >
                      0
                    </th>
                    <td
                      class="rt-TableCell"
                      data-testid="Ioo Gryffindor fund"
                    >
                      Ioo Gryffindor fund
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      fund manager
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      Singapore
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2000-07-06
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2024-02-21
                    </td>
                    <td
                      class="rt-TableCell"
                      data-testid="Infrastructure"
                    >
                      Infrastructure
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      15000000
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      GBP
                    </td>
                  </tr>
                  <tr
                    class="rt-TableRow"
                  >
                    <th
                      class="rt-TableCell rt-TableRowHeaderCell"
                      scope="row"
                    >
                      1
                    </th>
                    <td
                      class="rt-TableCell"
                      data-testid="Ibx Skywalker ltd"
                    >
                      Ibx Skywalker ltd
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      asset manager
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      United States
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      1997-07-21
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2024-02-21
                    </td>
                    <td
                      class="rt-TableCell"
                      data-testid="Infrastructure"
                    >
                      Infrastructure
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      31000000
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      GBP
                    </td>
                  </tr>
                  <tr
                    class="rt-TableRow"
                  >
                    <th
                      class="rt-TableCell rt-TableRowHeaderCell"
                      scope="row"
                    >
                      2
                    </th>
                    <td
                      class="rt-TableCell"
                      data-testid="Cza Weasley fund"
                    >
                      Cza Weasley fund
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      wealth manager
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      United Kingdom
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2002-05-29
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2024-02-21
                    </td>
                    <td
                      class="rt-TableCell"
                      data-testid="Hedge Funds"
                    >
                      Hedge Funds
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      58000000
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      GBP
                    </td>
                  </tr>
                  <tr
                    class="rt-TableRow"
                  >
                    <th
                      class="rt-TableCell rt-TableRowHeaderCell"
                      scope="row"
                    >
                      3
                    </th>
                    <td
                      class="rt-TableCell"
                      data-testid="Mjd Jedi fund"
                    >
                      Mjd Jedi fund
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      bank
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      China
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2010-06-08
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2024-02-21
                    </td>
                    <td
                      class="rt-TableCell"
                      data-testid="Private Equity"
                    >
                      Private Equity
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      72000000
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      GBP
                    </td>
                  </tr>
                  <tr
                    class="rt-TableRow"
                  >
                    <th
                      class="rt-TableCell rt-TableRowHeaderCell"
                      scope="row"
                    >
                      4
                    </th>
                    <td
                      class="rt-TableCell"
                      data-testid="Mjd Jedi fund"
                    >
                      Mjd Jedi fund
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      bank
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      China
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2010-06-08
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2024-02-21
                    </td>
                    <td
                      class="rt-TableCell"
                      data-testid="Natural Resources"
                    >
                      Natural Resources
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      1000000
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      GBP
                    </td>
                  </tr>
                  <tr
                    class="rt-TableRow"
                  >
                    <th
                      class="rt-TableCell rt-TableRowHeaderCell"
                      scope="row"
                    >
                      5
                    </th>
                    <td
                      class="rt-TableCell"
                      data-testid="Ibx Skywalker ltd"
                    >
                      Ibx Skywalker ltd
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      asset manager
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      United States
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      1997-07-21
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2024-02-21
                    </td>
                    <td
                      class="rt-TableCell"
                      data-testid="Real Estate"
                    >
                      Real Estate
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      17000000
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      GBP
                    </td>
                  </tr>
                  <tr
                    class="rt-TableRow"
                  >
                    <th
                      class="rt-TableCell rt-TableRowHeaderCell"
                      scope="row"
                    >
                      6
                    </th>
                    <td
                      class="rt-TableCell"
                      data-testid="Ibx Skywalker ltd"
                    >
                      Ibx Skywalker ltd
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      asset manager
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      United States
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      1997-07-21
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2024-02-21
                    </td>
                    <td
                      class="rt-TableCell"
                      data-testid="Real Estate"
                    >
                      Real Estate
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      83000000
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      GBP
                    </td>
                  </tr>
                  <tr
                    class="rt-TableRow"
                  >
                    <th
                      class="rt-TableCell rt-TableRowHeaderCell"
                      scope="row"
                    >
                      7
                    </th>
                    <td
                      class="rt-TableCell"
                      data-testid="Mjd Jedi fund"
                    >
                      Mjd Jedi fund
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      bank
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      China
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2010-06-08
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2024-02-21
                    </td>
                    <td
                      class="rt-TableCell"
                      data-testid="Hedge Funds"
                    >
                      Hedge Funds
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      28000000
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      GBP
                    </td>
                  </tr>
                  <tr
                    class="rt-TableRow"
                  >
                    <th
                      class="rt-TableCell rt-TableRowHeaderCell"
                      scope="row"
                    >
                      8
                    </th>
                    <td
                      class="rt-TableCell"
                      data-testid="Ibx Skywalker ltd"
                    >
                      Ibx Skywalker ltd
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      asset manager
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      United States
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      1997-07-21
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2024-02-21
                    </td>
                    <td
                      class="rt-TableCell"
                      data-testid="Hedge Funds"
                    >
                      Hedge Funds
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      85000000
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      GBP
                    </td>
                  </tr>
                  <tr
                    class="rt-TableRow"
                  >
                    <th
                      class="rt-TableCell rt-TableRowHeaderCell"
                      scope="row"
                    >
                      9
                    </th>
                    <td
                      class="rt-TableCell"
                      data-testid="Ioo Gryffindor fund"
                    >
                      Ioo Gryffindor fund
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      fund manager
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      Singapore
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2000-07-06
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      2024-02-21
                    </td>
                    <td
                      class="rt-TableCell"
                      data-testid="Hedge Funds"
                    >
                      Hedge Funds
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      31000000
                    </td>
                    <td
                      class="rt-TableCell"
                    >
                      GBP
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <div
            class="rt-ScrollAreaViewportFocusRing"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rt-TableRoot rt-r-size-1 rt-variant-ghost"
    >
      <div
        class="rt-ScrollAreaRoot"
        dir="ltr"
        style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
      >
        <style>
          [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
        </style>
        <div
          class="rt-ScrollAreaViewport"
          data-radix-scroll-area-viewport=""
          style="overflow-x: scroll; overflow-y: scroll;"
        >
          <div
            style="min-width: 100%; display: table;"
          >
            <table
              class="rt-TableRootTable rt-r-tl-auto"
            >
              <thead
                class="rt-TableHeader"
              >
                <tr
                  class="rt-TableRow"
                >
                  <th
                    align="left"
                    class="rt-TableCell rt-TableColumnHeaderCell"
                    scope="col"
                  >
                    ID
                  </th>
                  <th
                    align="left"
                    class="rt-TableCell rt-TableColumnHeaderCell"
                    scope="col"
                  >
                    Investor Name
                  </th>
                  <th
                    align="left"
                    class="rt-TableCell rt-TableColumnHeaderCell"
                    scope="col"
                  >
                    Investory Type
                  </th>
                  <th
                    align="left"
                    class="rt-TableCell rt-TableColumnHeaderCell"
                    scope="col"
                  >
                    Investor Country
                  </th>
                  <th
                    align="left"
                    class="rt-TableCell rt-TableColumnHeaderCell"
                    scope="col"
                  >
                    Investor Date Added
                  </th>
                  <th
                    align="left"
                    class="rt-TableCell rt-TableColumnHeaderCell"
                    scope="col"
                  >
                    Investor Last Updated
                  </th>
                  <th
                    align="left"
                    class="rt-TableCell rt-TableColumnHeaderCell"
                    scope="col"
                  >
                    Commitment Asset Class
                  </th>
                  <th
                    align="left"
                    class="rt-TableCell rt-TableColumnHeaderCell"
                    scope="col"
                  >
                    Commitmment Amount
                  </th>
                  <th
                    align="left"
                    class="rt-TableCell rt-TableColumnHeaderCell"
                    scope="col"
                  >
                    Commitmment Currency
                  </th>
                </tr>
              </thead>
              <tbody
                class="rt-TableBody"
              >
                <tr
                  class="rt-TableRow"
                >
                  <th
                    class="rt-TableCell rt-TableRowHeaderCell"
                    scope="row"
                  >
                    0
                  </th>
                  <td
                    class="rt-TableCell"
                    data-testid="Ioo Gryffindor fund"
                  >
                    Ioo Gryffindor fund
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    fund manager
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    Singapore
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2000-07-06
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2024-02-21
                  </td>
                  <td
                    class="rt-TableCell"
                    data-testid="Infrastructure"
                  >
                    Infrastructure
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    15000000
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    GBP
                  </td>
                </tr>
                <tr
                  class="rt-TableRow"
                >
                  <th
                    class="rt-TableCell rt-TableRowHeaderCell"
                    scope="row"
                  >
                    1
                  </th>
                  <td
                    class="rt-TableCell"
                    data-testid="Ibx Skywalker ltd"
                  >
                    Ibx Skywalker ltd
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    asset manager
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    United States
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    1997-07-21
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2024-02-21
                  </td>
                  <td
                    class="rt-TableCell"
                    data-testid="Infrastructure"
                  >
                    Infrastructure
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    31000000
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    GBP
                  </td>
                </tr>
                <tr
                  class="rt-TableRow"
                >
                  <th
                    class="rt-TableCell rt-TableRowHeaderCell"
                    scope="row"
                  >
                    2
                  </th>
                  <td
                    class="rt-TableCell"
                    data-testid="Cza Weasley fund"
                  >
                    Cza Weasley fund
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    wealth manager
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    United Kingdom
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2002-05-29
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2024-02-21
                  </td>
                  <td
                    class="rt-TableCell"
                    data-testid="Hedge Funds"
                  >
                    Hedge Funds
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    58000000
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    GBP
                  </td>
                </tr>
                <tr
                  class="rt-TableRow"
                >
                  <th
                    class="rt-TableCell rt-TableRowHeaderCell"
                    scope="row"
                  >
                    3
                  </th>
                  <td
                    class="rt-TableCell"
                    data-testid="Mjd Jedi fund"
                  >
                    Mjd Jedi fund
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    bank
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    China
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2010-06-08
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2024-02-21
                  </td>
                  <td
                    class="rt-TableCell"
                    data-testid="Private Equity"
                  >
                    Private Equity
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    72000000
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    GBP
                  </td>
                </tr>
                <tr
                  class="rt-TableRow"
                >
                  <th
                    class="rt-TableCell rt-TableRowHeaderCell"
                    scope="row"
                  >
                    4
                  </th>
                  <td
                    class="rt-TableCell"
                    data-testid="Mjd Jedi fund"
                  >
                    Mjd Jedi fund
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    bank
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    China
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2010-06-08
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2024-02-21
                  </td>
                  <td
                    class="rt-TableCell"
                    data-testid="Natural Resources"
                  >
                    Natural Resources
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    1000000
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    GBP
                  </td>
                </tr>
                <tr
                  class="rt-TableRow"
                >
                  <th
                    class="rt-TableCell rt-TableRowHeaderCell"
                    scope="row"
                  >
                    5
                  </th>
                  <td
                    class="rt-TableCell"
                    data-testid="Ibx Skywalker ltd"
                  >
                    Ibx Skywalker ltd
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    asset manager
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    United States
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    1997-07-21
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2024-02-21
                  </td>
                  <td
                    class="rt-TableCell"
                    data-testid="Real Estate"
                  >
                    Real Estate
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    17000000
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    GBP
                  </td>
                </tr>
                <tr
                  class="rt-TableRow"
                >
                  <th
                    class="rt-TableCell rt-TableRowHeaderCell"
                    scope="row"
                  >
                    6
                  </th>
                  <td
                    class="rt-TableCell"
                    data-testid="Ibx Skywalker ltd"
                  >
                    Ibx Skywalker ltd
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    asset manager
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    United States
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    1997-07-21
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2024-02-21
                  </td>
                  <td
                    class="rt-TableCell"
                    data-testid="Real Estate"
                  >
                    Real Estate
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    83000000
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    GBP
                  </td>
                </tr>
                <tr
                  class="rt-TableRow"
                >
                  <th
                    class="rt-TableCell rt-TableRowHeaderCell"
                    scope="row"
                  >
                    7
                  </th>
                  <td
                    class="rt-TableCell"
                    data-testid="Mjd Jedi fund"
                  >
                    Mjd Jedi fund
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    bank
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    China
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2010-06-08
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2024-02-21
                  </td>
                  <td
                    class="rt-TableCell"
                    data-testid="Hedge Funds"
                  >
                    Hedge Funds
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    28000000
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    GBP
                  </td>
                </tr>
                <tr
                  class="rt-TableRow"
                >
                  <th
                    class="rt-TableCell rt-TableRowHeaderCell"
                    scope="row"
                  >
                    8
                  </th>
                  <td
                    class="rt-TableCell"
                    data-testid="Ibx Skywalker ltd"
                  >
                    Ibx Skywalker ltd
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    asset manager
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    United States
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    1997-07-21
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2024-02-21
                  </td>
                  <td
                    class="rt-TableCell"
                    data-testid="Hedge Funds"
                  >
                    Hedge Funds
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    85000000
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    GBP
                  </td>
                </tr>
                <tr
                  class="rt-TableRow"
                >
                  <th
                    class="rt-TableCell rt-TableRowHeaderCell"
                    scope="row"
                  >
                    9
                  </th>
                  <td
                    class="rt-TableCell"
                    data-testid="Ioo Gryffindor fund"
                  >
                    Ioo Gryffindor fund
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    fund manager
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    Singapore
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2000-07-06
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    2024-02-21
                  </td>
                  <td
                    class="rt-TableCell"
                    data-testid="Hedge Funds"
                  >
                    Hedge Funds
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    31000000
                  </td>
                  <td
                    class="rt-TableCell"
                  >
                    GBP
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="rt-ScrollAreaViewportFocusRing"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
